### PR TITLE
Add 'Transfer between my accounts' checkbox and a respective field

### PR DESCRIPTION
**Pull request by keshan3262**

### CODE REVIEW

File: src/modals/send-modal/send-modal.tsx [hunk 1]
```diff
@@ -16,17 +17,25 @@ import { ScreenContainer } from '../../components/screen-container/screen-contai
...
+import {
+  useHdAccountsListSelector,
+  useSelectedAccountSelector,
+  useTezosTokenSelector,
+  useTokensListSelector
+} from '../../store/wallet/wallet-selectors';
 import { formatSize } from '../../styles/format-size';
 import { TEZ_TOKEN_METADATA } from '../../token/data/tokens-metadata';
 import { emptyToken, TokenInterface } from '../../token/interfaces/token.interface';
+import { conditionalStyle } from '../../utils/conditional-style';
```
Comment (andrewkryshtal) [inline src/modals/send-modal/send-modal.tsx:35]: unused import

File: src/modals/send-modal/send-modal.tsx [hunk 2]
```diff
@@ -36,6 +45,9 @@ export const SendModal: FC = () => {
   const { asset: initialAsset } = useRoute<RouteProp<ModalsParamList, ModalsEnum.Send>>().params;
   const { goBack } = useNavigation();
 
+  const { publicKeyHash: senderPublicKeyHash } = useSelectedAccountSelector();
+  const styles = useSendModalStyles();
+  const accounts = useHdAccountsListSelector();
```
Comment (unixvb) [inline src/modals/send-modal/send-modal.tsx:50]: filter accounts with the selected account.

File: src/modals/send-modal/send-modal.tsx [hunk 3]
```diff
@@ -36,6 +45,9 @@ export const SendModal: FC = () => {
   const { asset: initialAsset } = useRoute<RouteProp<ModalsParamList, ModalsEnum.Send>>().params;
   const { goBack } = useNavigation();
 
+  const { publicKeyHash: senderPublicKeyHash } = useSelectedAccountSelector();
```
Comment (unixvb) [inline src/modals/send-modal/send-modal.tsx:48]: remove this destructuring please, just use `selectedAccount. publicKeyHash` instead

File: src/modals/send-modal/send-modal.tsx [hunk 4]
```diff
@@ -73,6 +93,8 @@ export const SendModal: FC = () => {
       validationSchema={sendModalValidationSchema}
       onSubmit={onSubmit}>
       {({ values, setFieldValue, submitForm }) => {
+        const { transferBetweenOwnAccounts } = values;
```
Comment (unixvb) [inline src/modals/send-modal/send-modal.tsx:96]: the same with object destructuring here. `values.transferBetweenOwnAccounts`

File: src/components/checkbox/checkbox.tsx [hunk 5]
```diff
@@ -5,39 +5,55 @@ import { Svg, Path, Rect } from 'react-native-svg';
...
         />
-      )}
-    </Svg>
-    {children}
-  </TouchableOpacity>
-);
+        {value && (
+          <Path
+            d="M7 12.75L10 15.75L17.5 8"
+            stroke="#FF5B00"
```
Comment (unixvb) [inline src/components/checkbox/checkbox.tsx:49]: reuse `orange` variable

File: src/components/checkbox/checkbox.tsx [hunk 6]
```diff
@@ -5,39 +5,55 @@ import { Svg, Path, Rect } from 'react-native-svg';
...
+      hitSlop={generateHitSlop(formatSize(4))}
+      onPress={() => onChange(!value)}>
+      <Svg width={size} height={size} viewBox="0 0 24 24" fill="none">
+        <Rect
+          x="2.75"
+          y="2.75"
+          width="18.5"
+          height="18.5"
+          rx="2.25"
+          stroke={disabled ? colors.gray1 : '#FF5B00'}
```
Comment (unixvb) [inline src/components/checkbox/checkbox.tsx:43]: reuse `orange` variable

File: src/components/checkbox/checkbox.tsx (hunk 5 above)
Comment (unixvb) [reply src/components/checkbox/checkbox.tsx:49]: the same with `disabled ? colors.gray1 : '#FF5B00'`?

File: src/form/form-checkbox.tsx [hunk 7]
```diff
@@ -18,7 +20,7 @@ export const FormCheckbox: FC<Props> = ({ name, children }) => {
 
   return (
     <>
-      <Checkbox value={field.value} onChange={handleChange}>
+      <Checkbox disabled={disabled} value={field.value} onChange={handleChange} size={size}>
```
Comment (unixvb) [inline src/form/form-checkbox.tsx:23]: `onChange` should be the last

File: src/components/checkbox/checkbox.tsx [hunk 8]
```diff
@@ -5,39 +5,55 @@ import { Svg, Path, Rect } from 'react-native-svg';
...
   onChange: EventFn<boolean>;
 }
 
 export const Checkbox: FC<Props> = ({
+  disabled,
   value,
   size = formatSize(24),
   strokeWidth = formatSize(1.5),
   children,
   onChange
```
Comment (unixvb) [inline src/components/checkbox/checkbox.tsx:25]: Try to keep the order of props the same overywhere
Screenshot 2021-07-16 at 16 28 48
image

**Review by unixvb — APPROVED** [review]

**Review by andrewkryshtal — APPROVED** [review]

**Pull request merged**